sdk/identity/azure-identity/azure/identity/_credentials/default.py

```diff
@@ -256,3 +276,18 @@ def get_token_info(self, *scopes: str, options: Optional[TokenRequestOptions] =
         token_info = cast(SupportsTokenInfo, super()).get_token_info(*scopes, options=options)
         within_dac.set(False)
         return token_info
+
+
+def parse_azure_dac(az_dac, avail_credentials):
+    striped_az_dac = az_dac.strip()
+    if striped_az_dac.endswith(";"):
+        striped_az_dac = striped_az_dac[:-1]
+    creds = [cred.strip() for cred in striped_az_dac.split(";")]
+    credentials = []
+
+    for cred in creds:
+        if cred not in VALID_CREDENTIALS:
+            raise ValueError(f"Invalid credential '{cred}' in AZURE_DEFAULT_CREDENTIAL_ALLOW_LIST")
```

Comment:
Might be helpful to include the valid list of credential strings in the error.

```suggestion
            raise ValueError(
                f"Invalid credential '{cred}' in AZURE_DEFAULT_CREDENTIAL_ALLOW_LIST. "
                f"Valid credentials are {', '.join(VALID_CREDENTIALS)}."
            )
```